### PR TITLE
[HDInsight] - Support ADLS Gen2 MSI

### PR DESCRIPTION
**Pull request by idear1203**
## Note

- Update `azure-mgmt-hdinsight` SDK version to 0.2.1
- `az hdinsight create`: added the `--storage-account-managed-identity` parameter to support ADLS Gen2 MSI.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py [hunk 1]
```diff
@@ -99,13 +100,13 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
         raise CLIError('Either the default container or the default filesystem can be specified, but not both.')
 
     # Retrieve primary blob service endpoint
+    is_wasb = True if not storage_account_managed_identity else False
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py]: maybe just `is_wasb = not storage_account_managed_identity`?

File: src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py [hunk 2]
```diff
@@ -115,16 +116,20 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
...
     # Validate storage info parameters
-    if not _all_or_none(storage_account, storage_account_key,
-                        (storage_default_container or storage_default_filesystem)):
+    if is_wasb and not _all_or_none(storage_account, storage_account_key, storage_default_container):
         raise CLIError('If storage details are specified, the storage account, storage account key, '
-                       'and either the default container or default filesystem must be specified.')
+                       'and the default container must be specified.')
+    elif not is_wasb and not _all_or_none(storage_account, storage_default_filesystem):
+        raise CLIError('If storage details are specified, the storage account, '
+                       'and the default filesystem must be specified.')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py]: please double check whether or not the error text is right. Not so sure what `_all_or_none` means really.

File: src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py [hunk 3]
```diff
@@ -202,6 +207,8 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
                 key=storage_account_key,
                 container=storage_default_container,
                 file_system=storage_default_filesystem,
+                resource_id=storage_account if not is_wasb else None,
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py]: Suggest `None if is_wasb else storage_account` to get rid of the `not`

File: src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py (hunk 1 above)
Comment (idear1203) [reply src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py]: Thanks for advance! I will update the logic.

File: src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py (hunk 2 above)
Comment (idear1203) [reply src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py]: Good catch! This logic is not quite accurate but cannot really be reached. I will make a little change to check none only.

File: src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py (hunk 3 above)
Comment (idear1203) [reply src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py]: Good suggestion! I am still on my way of learning Python:)

File: src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py (hunk 2 above)
Comment (idear1203) [reply src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py]: After a third thought, this is correct logic. Let's keep it as what it is now.

**Pull request merged**